flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/AsyncCheckpointRunnable.java

```diff
@@ -119,10 +119,7 @@ public void run() {
 
```

Comment:
Can you copy paste the PR description into the commit message?
> This PR fixes the issue that when a task finished on restore taking unaligned checkpoints, currently the asynchronous part does not wait till the alignment of barriers get finished, thus throws exception due to some metrics are not set yet.

+ maybe copy/paste from the JIRA ticket

```
. The task received the first barrier.
2. With the process of unaligned checkpoint, the task would snapshot the state of the operators.
3. The checkpoint would start the asynchronous part.
4. Normally in the asynchronous part, it would wait till all the state futures get done, including the channel states and result partition states. With this method, it ensures the asynchronous part would wait till the last barrier arrived. But if the task has been fully finished before, these states are ignored and the assumption is broken.
5. Then the asynchronous part would fail since when it try to build the CheckpointMetrics, the alignment for this checkpoint is in fact not done yet.
```
?